src/main/java/spoon/support/compiler/jdt/TreeBuilderCompiler.java

```diff
@@ -72,6 +72,8 @@
 			requestor.acceptResult(unit.compilationResult);
 		}
 
-		return this.unitsToProcess;
+		CompilationUnitDeclaration[] unitsToReturn = new CompilationUnitDeclaration[this.totalUnits];
+		System.arraycopy(this.unitsToProcess, 0, unitsToReturn, 0, this.totalUnits);
```

Comment:
That seems more smart to iterate on `unitsToProcess` to check which items are `null` and returns this list.